molo/core/api/importers.py

```diff
@@ -0,0 +1,231 @@
+        self._base_url = base_url
+
+    def get_content_from_url(self, base_url):
+        """
+        Sections can have SectionPage and ArticlePage child objects.
+        These have different fields, and thus have to be treated
+        differently.
+        """
+        # assemble url
+        base_url = base_url.rstrip("/")
+        url = base_url + API_PAGES_ENDPOINT + "?type=" + self._content_type + \
+            "&fields=" + ",".join(self._fields) + \
+            "&order=latest_revision_created_at"
+
+        # make request
+        try:
+            response = requests.get(url)
+            self._base_url = base_url
+            self._content = response.json()
+            self._content = self._content["items"]
+            return self._content
+        except requests.exceptions.ConnectionError:
+            return "No content could be found from {}. " \
+                "Are you sure this is the correct URL?".format(base_url)
+        except requests.exceptions.RequestException:
+            return "Content could not be imported at this time. " \
+                   "Please try again later."
+
+    def content(self):
+        if self._content:
+            return self._content
+        return []
+
+    def save(self, indexes, parent_id):
+        pass
+
+
+class ArticlePageImporter(PageImporter):
+
+    def __init__(self, base_url=None, content=None, content_type=None):
+        super(ArticlePageImporter, self).__init__(
+            base_url=base_url, content=content, content_type=content_type
+        )
+        self._content_type = "core.ArticlePage"
+        self._fields = ArticlePage.get_api_fields()
+
+    def save(self, indexes, parent_id):
+        if self.content():
+            parent = Page.objects.get(id=parent_id)
+            for index in indexes:
+                # Remove "id" and "meta" fields
+                selected_article = self.content()[index]
+                fields, nested_fields = separate_fields(selected_article)
+                article = ArticlePage(**fields)
+
+                # TODO: u'related_sections'
+                # process the nested fields
+                if ("tags" in nested_fields) and nested_fields["tags"]:
+                    article.tags.add(", ".join(nested_fields["tags"]))
+
+                if ("metadata_tags" in nested_fields) and \
+                        nested_fields["metadata_tags"]:
+                    article.metadata_tags.add(
+                        ", ".join(nested_fields["metadata_tags"])
+                    )
+
+                if ("body" in nested_fields) and nested_fields["body"]:
+                    article.body = json.dumps(nested_fields["body"])
+
+                if ("image" in nested_fields) and nested_fields["image"]:
+                    article.image = get_image(
+                        self._base_url, nested_fields["image"]["id"]
+                    )
+
+                parent.add_child(instance=article)
+                parent.save_revision().publish()
```

Comment:
This assumes that the parent was originally published. We need to handle the case where not all the articles/sections on the source site are published. They will need to be imported, but their published status will need to be respected.